src/transforms/push.ts

```diff
@@ -152,15 +155,18 @@ export const processPush = async (github: GitHubAPI, payload, rootLogger: Logger
 		const commits: JiraCommit[] = await Promise.all(
 			shas.map(async (sha): Promise<JiraCommit> => {
 				log.info("Calling GitHub to fetch commit info " + sha.id);
+				const useNewGithubClient = await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_PUSH, false, subscription.jiraHost);
 				try {
 					const {
 						data,
 						data: {commit: githubCommit},
-					} = await github.repos.getCommit({
-						owner: owner.login,
-						repo,
-						ref: sha.id,
-					});
+					} = useNewGithubClient
+						? await github.getCommit(owner.login, repo, sha.id)
+						: await githubOld.repos.getCommit({
+							owner: owner.login,
+							repo,
+							ref: sha.id,
+						});
```

Comment:
This is the main change. Using the new GitHub client to get a commit on the push path. This is what caused the queue to clog up last night.